contracts/base/BaseLiquidityManagement.sol

```diff
@@ -126,66 +126,45 @@ contract BaseLiquidityManagement is IBaseLiquidityManagement, SafeCallback {
-        // If the totalFeesAccrued equals the callerFeesAccrued then the total owed to the caller is just the liquidityDelta.
-        // If the totalFeesAccrued is greater than the callerFeesAccrued, we must account for the difference.
-        // TODO: If totalFeesAccrued == callerFeesAccrued, I think we can just apply the entire delta onto the caller, even if this implicitly collects on behalf of another user in the same range.
-        (int128 callerDelta0, int128 callerDelta1) = totalFeesAccrued != callerFeesAccrued
-            ? _calculateCallerDeltas(liquidityDelta, totalFeesAccrued, callerFeesAccrued)
-            : (liquidityDelta.amount0(), liquidityDelta.amount1());
+        // the delta for increase liquidity assuming that totalFeesAccrued was not applied
+        BalanceDelta principalDelta = liquidityDelta - totalFeesAccrued;
+
+        // outstanding deltas the caller is responsible for, after their fees are credited to the principal delta
+        BalanceDelta callerDelta = principalDelta + callerFeesAccrued;
+
+        // outstanding deltas this contract is responsible for, intuitively the contract is responsible for taking fees external to the caller's accrued fees
+        BalanceDelta thisDelta = totalFeesAccrued - callerFeesAccrued;
 
         // Update position storage, flushing the callerDelta value to tokensOwed first if necessary.
         // If callerDelta > 0, then even after investing callerFeesAccrued, the caller still has some amount to collect that were not added into the position so they are accounted to tokensOwed and removed from the final callerDelta returned.
-        uint128 tokensOwed0 = 0;
-        uint128 tokensOwed1 = 0;
-        (tokensOwed0, callerDelta0) = callerDelta0 > 0 ? (uint128(callerDelta0), int128(0)) : (uint128(0), callerDelta0);
-        (tokensOwed1, callerDelta1) = callerDelta1 > 0 ? (uint128(callerDelta1), int128(0)) : (uint128(0), callerDelta1);
+        BalanceDelta tokensOwed;
+        if (callerDelta.amount0() > 0) {
+            // credit the excess tokens to the position's tokensOwed
+            tokensOwed = toBalanceDelta(callerDelta.amount0(), 0);
 
-        position.addTokensOwed(tokensOwed0, tokensOwed1);
-        position.addLiquidity(liquidityToAdd);
-        position.updateFeeGrowthInside(feeGrowthInside0X128, feeGrowthInside1X128);
+            // this contract is responsible for custodying the excess tokens
+            thisDelta = thisDelta + toBalanceDelta(callerDelta.amount0(), 0);
 
-        // The delta owed or credited by this contract.
-        // TODO @sauce check that if callerDelta == 0 (zerod out from above), then this line just credits the posm to takes on behalf of the caller
-        int128 thisDelta0 = liquidityDelta.amount0() - callerDelta0;
-        int128 thisDelta1 = liquidityDelta.amount1() - callerDelta1;
+            // the caller is not expected to collect the excess tokens
+            callerDelta = toBalanceDelta(0, callerDelta.amount1());
```

Comment:
since all of this is shared once for token0 and once for token1 can we just put it in a helper?